indexer/documents/conv.go

```diff
@@ -88,17 +88,27 @@ func ConvTx(txIdx uint64, tx *types.Tx, receipt *types.Receipt, blockDoc *EsBloc
 	}
 }
 
-// ConvContractCreateTx creates document for token creation
-func ConvContract(txDoc *EsTx, contractAddress []byte) *EsContract {
-
+// ConvContractFromTx creates document for contract creation
+func ConvContractFromTx(txDoc *EsTx, contractAddress []byte) *EsContract {
 	byteCode, sourceCode, abi, deployArgs := extractContractCode(txDoc.Payload)
+	return ConvContract(txDoc, contractAddress, txDoc.Account, byteCode, abi, sourceCode, deployArgs)
+}
+
+func ConvContractFromCall(txDoc *EsTx, contractAddress []byte, creator, sourceCode, deployArgs string) *EsContract {
+	byteCode, abi, err := CompileSourceCode(sourceCode)
+	if err != nil {
+		panic(err)
```

Comment:
Is it safe to invoke panic here? Does any caller catches the panic?

Reply:
I don't know. The idea is to make the app to close
What would be the alternative? Maybe retry?

Reply:
Neither do I. You know much more than me about indexer. If panic is propagated to top level of call stack, the process will be terminated. I don't know this is expected behavior or not of indexer.